site/src/app/v2/trouver-un-club/page.tsx

```diff
@@ -4,32 +4,55 @@ import { Card } from '@codegouvfr/react-dsfr/Card';
 import { Tag } from '@codegouvfr/react-dsfr/Tag';
 import styles from './style.module.scss';
 import PageHeader from '../../../../components/PageHeader/PageHeader';
-import { useEffect, useState } from 'react';
+import { useEffect, useMemo, useState } from 'react';
 import Search from './components/search/Search';
 import { SqlSearchParams, getClubs } from './agent';
 import { usePathname } from 'next/navigation';
 import EligibilityTestBanner from './components/eligibilityTestBanner/EligibilityTestBanner';
+import Button from '@codegouvfr/react-dsfr/Button';
 
 export default function TrouverUnClub() {
+  const limit = 20;
   const pathName = usePathname();
+
   const [clubs, setClubs] = useState<SportGouvJSONResponse | undefined>();
-  const [clubParams, setClubParams] = useState<SqlSearchParams | undefined>({});
+  const [clubParams, setClubParams] = useState<SqlSearchParams>({ limit, offset: 0 });
+
+  const { nom, offset } = clubParams;
 
   useEffect(() => {
-    getClubs(clubParams).then((res) => setClubs(res));
-  }, [clubParams]);
+    if (offset === 0) {
+      getClubs({ nom, offset, limit }).then((res) => setClubs(res));
+    } else {
+      getClubs({ nom, offset, limit }).then((res) =>
+        setClubs((clubs) => {
+          if (clubs?.results) {
+            return { results: [...clubs.results, ...res.results], total_count: res.total_count };
+          } else {
+            return res;
+          }
```

Comment:
Privilégie le early return, c'est plus lisible
```suggestion
          if (!clubs.results) {
            return res;
          }

          return { results: [...clubs.results, ...res.results], total_count: res.total_count };
```

Reply:
cette partie s'est simplifiée